src/components/TodoFilter/TodoFilter.tsx

```diff
@@ -1,14 +1,46 @@
 import React from 'react';
+import { useAppDispatch, useAppSelector } from '../../app/hooks';
+import { Status } from '../../types/Status';
+import { actions as filterActions } from '../../features/filter';
 
 export const TodoFilter: React.FC = () => {
+  const dispatch = useAppDispatch();
+  const { query, status } = useAppSelector(state => state.filter);
+
+  const setStatus = (value: Status) => dispatch(filterActions.setStatus(value));
+  const setQuery = (queryValue: string) => (
+    dispatch(filterActions.setQuery(queryValue))
+  );
+  const clearQuery = () => dispatch(filterActions.setQuery(''));
+
+  const handleStatusChange = (event: React.ChangeEvent<HTMLSelectElement>) => {
+    const selectedStatusValue = event.target.value as Status;
+
+    if (status === selectedStatusValue) {
```

Comment:
You are returning from the function if the selected status is the same as the current status. This is not necessary, as setting the state with the same value does not cause a re-render or any side effects.